components/script/dom/elementinternals.rs

```diff
@@ -208,6 +216,7 @@ impl ElementInternalsMethods for ElementInternals {
         // Step 4: For each entry `flag` → `value` of `flags`, set element's validity flag with the name
         // `flag` to `value`.
         self.validity_state().update_invalid_flags(bits);
+        self.validity_state().update_pseudo_classes();
```

Comment:
After `update_invalid_flags`, we should `update_pseudo_classes`, so that the values of `:valid` and `:invalid` pseud classes are updated.